### PR TITLE
Course Copy

### PR DESCRIPTION
**Pull request by brianlugo1**
Tasks Completed:
- Implemented assessment_task_copy route!
- Implemented AdminImportAssessmentTasks.js component!
- Implemented CourseDropdown.js component!
- Updated AdminDashboard to Link Import Task button!
- Implemented a ImportAssessmentTask page in Navbar.js component!

### CODE REVIEW

**Comment by sah0017** [comment]
Task linked: [CU-8678tu58d This was a duplicate:  As an administrator, so I can reuse my assessment task setup from one semester to the next or for multiple sections, I want a course copy feature.](https://app.clickup.com/t/8678tu58d)

File: FrontEndReact/src/View/Admin/Add/ImportTasks/AdminImportAssessmentTasks.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                    <div className="d-flex justify-content-around">
+                        Please select the course you would like to import assesments tasks from
+                    </div>
+                    <div className="d-flex flex-column">
+                        <div className="d-flex flex-row justify-content-between">
+                            <div className="w-25 p-2 justify-content-between">
+                                <label id="dueDateLabel">Course</label>
+                            </div>
+                            <div className="d-flex flex-row justify-content-around">
+                                {/* Add a dropdown for selecting courses here */}
```
Comment (Wrimo) [inline FrontEndReact/src/View/Admin/Add/ImportTasks/AdminImportAssessmentTasks.js:108]: I believe this comment is out of date.

File: FrontEndReact/src/View/Admin/Add/ImportTasks/AdminImportAssessmentTasks.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            });
+        };
+    }
+    componentDidMount() {
+        document.getElementById("importAssessmentTasks").addEventListener("click", () => {
+            var message = "Invalid Form: ";
+            if(!validator.isNumeric(String(this.state.selectedCourse)) && validator.equals(this.state.selectedCourse, '')) {
+                message += "Missing Course!";
+            }
+            if(message === "Invalid Form: ") {
```
Comment (Wrimo) [inline FrontEndReact/src/View/Admin/Add/ImportTasks/AdminImportAssessmentTasks.js:30]: I know this is the standard method in the front end that's been used for checking if a form is invalid, but I really don't like it. The intention isn't clear (a valid form happens when the message is "Invalid") and it's not great practice to use string comparison for this sort of logic.

File: FrontEndReact/src/View/Admin/Add/ImportTasks/AdminImportAssessmentTasks.js (hunk 1 above)
Comment (brianlugo1) [reply FrontEndReact/src/View/Admin/Add/ImportTasks/AdminImportAssessmentTasks.js:108]: Comment removed!

File: FrontEndReact/src/View/Admin/Add/ImportTasks/AdminImportAssessmentTasks.js (hunk 2 above)
Comment (brianlugo1) [reply FrontEndReact/src/View/Admin/Add/ImportTasks/AdminImportAssessmentTasks.js]: Fixed! I have create a variable called success to clearly indicate whether an error has occurred with form validation!

**Review by Wrimo — APPROVED** [review]
LGTM

**Review by aldiux4you — APPROVED** [review]
LGTM! Some of the components and CSS will be updated once we merge into main.

**Pull request merged**